prow/spyglass/gcsartifact_fetcher.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/option"
+
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+// A fetcher for a GCS client
+type GCSArtifactFetcher struct {
+	Client      *storage.Client
+	XMLEndpoint string
+	WithTLS     bool
+}
+
+// A location in GCS where Prow job-specific artifacts are stored. This implementation assumes
+// Prow's native GCS upload format (treating GCS keys as a directory structure), and is not
+// intended to support arbitrary GCS bucket upload formats.
+type GCSJobSource struct {
+	source     string
+	linkPrefix string
+	bucket     string
+	jobPath    string
+	jobName    string
+	buildID    string
+}
+
+// NewGCSArtifactFetcher creates a new ArtifactFetcher with a real GCS Client
+func NewGCSArtifactFetcher() *GCSArtifactFetcher {
+	c, err := storage.NewClient(context.Background(), option.WithoutAuthentication())
+	if err != nil {
+		log.Fatal(err)
```

Comment:
create the client at the top-level entrypoint to `deck` and pass it in

Reply:
please put a `// TODO` at the GCS client construction about potential future support for authentication